server/server.go

```diff
@@ -67,35 +55,36 @@ func (s *Server) create(successCallback func([]beat.Event), host string) *http.S
 		})
 	}
 
-	return &http.Server{
+	ser := &http.Server{
 		Addr:           host,
 		Handler:        mux,
-		ReadTimeout:    s.config.ReadTimeout,
-		WriteTimeout:   s.config.WriteTimeout,
-		MaxHeaderBytes: s.config.MaxHeaderBytes,
+		ReadTimeout:    config.ReadTimeout,
+		WriteTimeout:   config.WriteTimeout,
+		MaxHeaderBytes: config.MaxHeaderBytes,
 	}
+
+	return &ApmServer{config: config, httpServer: ser}
 }
 
-func (s *Server) Start(successCallback func([]beat.Event), host string) {
-	s.http = s.create(successCallback, host)
-	if s.config.SSL.IsEnabled() {
-		go s.http.ListenAndServeTLS(s.config.SSL.Cert, s.config.SSL.PrivateKey)
+func (apm *ApmServer) Start() {
+	if apm.config.SSL.IsEnabled() {
+		go apm.httpServer.ListenAndServeTLS(apm.config.SSL.Cert, apm.config.SSL.PrivateKey)
 	} else {
-		go s.http.ListenAndServe()
+		go apm.httpServer.ListenAndServe()
 	}
 }
 
-func (s *Server) Stop() error {
+func (apm *ApmServer) Stop() error {
 	c := context.TODO()
-	err := s.http.Shutdown(c)
+	err := apm.httpServer.Shutdown(c)
 	return err
 }
 
-func (s *Server) createHandler(p processor.Processor, successCallback func([]beat.Event)) func(w http.ResponseWriter, r *http.Request) {
+func createHandler(p processor.Processor, config Config, successCallback func([]beat.Event)) func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I actually added this to server in the past exactly to be able to use the `config` and not have to pass it as param.